### PR TITLE
Add building docs to CI checks

### LINKED ISSUE #69: Add building docs to automated tests that run on github
Would be nice to have the CI check that docs can build without error

### PR DESCRIPTION
**Pull request by gsheni**
- Fixes #69
- Added `[jupyter]` add-on for `black` to standardize notebook cells
- Updated `black` to the latest
- Added `make docs` command in root directory to build docs

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -51,14 +44,12 @@ dev = [
...
     "pre-commit == 2.20.0",
-    "gridstatus[dev]",
 ]
-
 docs = [
     "ipython == 8.4.0",
     "jupyter == 1.0.0",
     "matplotlib == 3.5.2",
-    "sphinx~=4.0",
+    "sphinx == 4.5.0",
```
Comment (gsheni) [inline pyproject.toml:52]:
```suggestion
    "sphinx~=4.0",
```

File: .github/workflows/docs.yaml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+      - name: Checkout repository
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.pull_request.head.ref }}
+          repository: ${{ github.event.pull_request.head.repo.full_name }}
+      - name: Build gridstatus package
+        run: make package
+      - name: Install gridstatus 
+        run: |
+          python -m pip install unpacked/[docs]
```
Comment (gsheni) [inline .github/workflows/docs.yaml:29]:
```suggestion
          python -m pip install "unpacked/[docs]"
```

**Comment by gsheni** [comment]
@kmax12 Any suggestions on how to fix the autoapi error?
```
/home/runner/work/gridstatus/gridstatus/docs/_auto_api_templates/python/attribute.rst:4:Unexpected indentation.
```
- I've tried to loosen the doc requirements, and modify the *.rst files with no luck.

**Comment by kmax12** [comment]
hmm...I dont know off top of my head. will have to take a look

File: .github/workflows/docs.yaml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+        with:
+          ref: ${{ github.event.pull_request.head.ref }}
+          repository: ${{ github.event.pull_request.head.repo.full_name }}
+      - name: Build gridstatus package
+        run: make package
+      - name: Install gridstatus
+        run: |
+          python -m pip install "unpacked/[docs]"
+      - name: Build docs
+        run: make -C docs/ -e "SPHINXOPTS=-j auto" clean html
```
Comment (gsheni) [inline .github/workflows/docs.yaml:31]: There is no `SPHINXOPTS="-W"` here which means that warnings will NOT be treated as errors.

**Comment by gsheni** [comment]
@kmax12 This MR is ready for review. 
I'm not sure why the tests are failing, as I didn't modify any files in the actual library. My guess is the API endpoints are down.

File: Makefile [hunk 4]
```diff
@@ -46,3 +55,7 @@ package: upgradepip upgradebuild upgradesetuptools
 	$(eval PACKAGE=$(shell python -c "from pep517.meta import load; metadata = load('.'); print(metadata.version)"))
 	tar -zxvf "dist/gridstatus-${PACKAGE}.tar.gz"
 	mv "gridstatus-${PACKAGE}" unpacked
+
+.PHONY: docs
+docs: clean
```
Comment (kmax12) [inline Makefile:60]: i believe this is a command to build the docs from the root of the project. can you update contributing.md to mention this command? 

the 3 item in the section on building the docs

image

**Review by kmax12 — COMMENTED** [review]
Looks good. just one comment then ready to merge. Thank you!

File: Makefile (hunk 4 above)
Comment (gsheni) [reply Makefile:60]: Sure, fixed:
https://github.com/kmax12/gridstatus/pull/71/commits/7331383f56e039ede47bc14db6764fda479527fd

**Review by kmax12 — APPROVED** [review]

**Pull request merged**